### ISSUE TITLE
Relaxed Loss

### ISSUE THREAD

**Issue by mahd7395** [NVIDIA/semantic-segmentation#80]
Hi
I was using your relaxed loss and I'm just getting this error, I don't know what should I do?
Function 'LogBackward' returned nan values in its 0th output

**Comment by karansapra** [comment]
Hi, 

Sorry its been a while since I have tested that code path since we were working on our new SOTA work, i believe they change dim order that might be causing an issue. I will see if i can find some time to test out the path and create unit_test. I would recommend going to an old version of pytorch and trying it out.

**Comment by mahd7395** [comment]
Thanks for your response.
I've tried every older version of PyTorch, but I'm still getting this error.
Actually this line return a nan value :
loss.py, line 177, in custom_nll
customsoftmax(inputs, target[:, :-1, :, :].float())).sum(1)) .
loss.py, line 121, in customsoftmax.
torch.max(soft, (multihotmask * (soft * multihotmask).sum(1, keepdim=True))).

**Comment by CheeseInTrap** [comment]
@mahd7395 Did you find any solution? I encountered the same problem

**Comment by mahd7395** [comment]
@CheeseInTrap Actually no, I didn't find any solution.

**Comment by karansapra** [comment]
Hi @mahd7395 @CheeseInTrap  to work we had to modify the labels to be relaxed. I wonder if we disable that path of the code. Sorry i haven't had time to look at this because of other work. I ll try find some time maybe this weekend or early next week. But the intuition is that there is a modification of the input labels so I am now wondering if the root cause, i.e. that it's not hitting that path of the code where it relaxes the label in the dataloader.


Have you guys tried or already been working with this branch ? 
https://github.com/NVIDIA/semantic-segmentation/tree/sdcnet

**Comment by mahd7395** [comment]
@karansapra yea I run the code from this  https://github.com/NVIDIA/semantic-segmentation/tree/sdcnet.
I know your intuition for the modification of labels space. it's brilliant.
I have another question. how can I use your relaxed loss when my input tensors are 5D, I mean when tensors are like this (B, C, D, W, L).

**Comment by karansapra** [comment]
Hmm, i havent thought it through but wouldn't reshape apply the loss and reshape work ? By adding D in the dimension you are just adding per channel-wise softmax, right ?

**Comment by mahd7395** [comment]
actually, I have a video dataset so I have a sequence of frames. D represents the number of frames.

**Comment by karansapra** [comment]
So you are softening labels per image right ? or post warped images that you want to soften labels over to produce one single label for D set of frames ?

**Comment by mahd7395** [comment]
yes, I'm softening labels per image.

**Comment by mahd7395** [comment]
@karansapra did you check why relaxed loss return NAN?

**Comment by karansapra** [comment]
Hi @mahd7395 i haven't checked, sorry we have been busy with some upcoming deadlines. I will try get it ASAP, i have it on my list to do. Is post CVPR ok with you ?
